examples/plot_firing_pattern.py

```diff
@@ -30,26 +29,33 @@
 import matplotlib.pyplot as plt
 
 net = Network(params)
-with NetworkBuilder(net) as network_builder:
-    network_builder.cells[0].plot_voltage()
-
-    # The cells are stored in the network object as a list
-    cells = network_builder.cells
-    print(cells[:5])
-
-    # We have different kinds of cells with different cell IDs (gids)
-    gids = [0, 35, 135, 170]
-    for gid in gids:
-        print(cells[gid].name)
-
-    # We can plot the firing pattern of individual cells
-    network_builder.cells[0].plot_voltage()
-    plt.title('%s (gid=%d)' % (cells[0].name, gid))
+dpls = simulate_dipole(net, n_trials=1)
+
+# The cell IDs (gids) are stored in the network object as a dictionary
+gid_dict = net.gid_dict
+print(net.gid_dict)
+
+# Simulated cell responses are stored in the Spikes object as a dictionary.
+trial_idx = 0
+vsoma = net.spikes.vsoma[trial_idx]
+print(vsoma.keys())
+
+# We can plot the firing pattern of individual cells by indexing with the gid
+gid = 170
+t_vec = net.spikes.t_vec[trial_idx]
+plt.figure(figsize=(4, 4))
+plt.plot(t_vec, vsoma[gid])
+plt.title('%s (gid=%d)' % (net.gid_to_type(gid), gid))
+plt.xlabel('Time (ms)')
+plt.ylabel('Voltage (mV)')
+plt.show()
 
 ###############################################################################
-# Let's do this for the rest of the cell types with a new NetworkBuilder object
-with NetworkBuilder(net) as network_builder:
-    fig, axes = plt.subplots(1, 2, sharey=True, figsize=(8, 4))
-    for gid, ax in zip([35, 170], axes):
-        network_builder.cells[gid].plot_voltage(ax)
-        ax.set_title('%s (gid=%d)' % (cells[gid].name, gid))
+# # Let's do this for the rest of the cell types
+fig, axes = plt.subplots(1, 2, sharey=True, figsize=(8, 4))
+for gid, ax in zip([35, 170], axes):
```

Comment:
curious where gids 35 and 170 came from @jasmainak ? if they are of a certain type, would it be useful to add that in a comment?

Reply:
sure, or perhaps use `gid_dict['L2_Pyrmaidal'][0]` or some such thing

Reply:
> sure, or perhaps use `gid_dict['L2_Pyrmaidal'][0]` or some such thing

This is excellent. Almost self-documenting code.

Reply:
> Almost self-documenting code.

Good evidence that hnn-core is improving! Also added this modification, looks like the numbers were chosen because they are the boundaries of the gid ranges which define the cell classes.